torch/csrc/autograd/function.h

```diff
@@ -56,7 +56,7 @@ struct MakeNextFunctionList : IterArgs<MakeNextFunctionList> {
 inline bool any_variable_requires_grad(const variable_list& variables) {
   return std::any_of(
       variables.begin(), variables.end(), [](const Variable& variable) {
-        return variable.requires_grad();
+        return variable.defined() && variable.requires_grad();
```

Comment:
I think this peoblem is already being addressed in another PR. Can you please revert it?

Reply:
Sure I will revert this.